### ISSUE TITLE
Error in CuDNN: CUDNN_STATUS_NOT_SUPPORTED (cudnnBatchNormalizationForwardInference)

### ISSUE THREAD

**Issue by binuq8usa** [soumith/cudnn.torch#318]
Hi, 

I am getting this error while training a residual Net-18 for image classification. This error happens whenever I set the batch size > 128. I am currently running the program on an NVidia GeForce GTX 1080. Any suggestions as to why this happens and possible solutions to try? This error does not occur for batch size <= 128. 

Thanks in advance.

**Comment by ngimel** [comment]
Inference mode for cudnn implementation of batch normalization has a limit for the batchsize (1024, IIRC)  - I'm not sure why you would be hitting it for smaller batch sizes. Anyway, this limit is no longer present in cudnn v6 rc. Also, you can use cunn implementation of batch norm that does not have this limitation.

**Comment by binuq8usa** [comment]
Thanks. I made the following change : 
cudnn.convert(net, cudnn, function(module)
   return torch.type(module):find('SpatialBatchNormalization')
end)
net = net:cuda()
In this case, every other module except for nn.SpatialBatchNormalization will be converted to cudnn. This works for a specific batch size ( batch_size = 128). Testing on different batch sizes and see if any other error comes up.

**Comment by binuq8usa** [comment]
A different error crops up now when i use batch size > 128. I will re-open the issue in another thread. Closing this one.
